store/tikv/2pc.go

```diff
@@ -339,11 +346,15 @@ func (c *twoPhaseCommitter) initKeysAndMutations() error {
 			lockCnt++
 		} else {
 			value = it.Value()
+			isUnnecessaryKV := filter != nil && filter.IsUnnecessaryKeyValue(key, value, flags)
 			if len(value) > 0 {
-				if tablecodec.IsUntouchedIndexKValue(key, value) {
+				if isUnnecessaryKV {
```

Comment:
Only un-empty value need the filter, so could we move the filter logic in L350~L351?